src/Dapr.Client/InvocationHandler.cs

```diff
@@ -128,17 +136,56 @@ internal bool TryRewriteUri(Uri? uri, [NotNullWhen(true)] out Uri? rewritten)
                 return false;
             }
 
+            string host;
+
+            // It is just for optimization, to not add some overhead time
+            if (this.DefaultAppId is not null && uri.Host.Equals(this.DefaultAppId, StringComparison.InvariantCultureIgnoreCase))
+            {
+                host = this.DefaultAppId;
+            }
+            else
+            {
+                host = this.GetOriginalHostFromUri(uri);
```

Comment:
I wonder if we really need to be concerned about this case; it's only if the caller of `DaprClient.CreateInvokeHttpClient()` does *not* provide an `appId`, right?  That use case is to have a "generic" HTTP client that can call any Dapr application and I wonder how often that's done vs. having a HTTP client that's specific to a specific Dapr application.

If the `appId` is omitted, then they would get the prior behavior (with attempts to use case-sensitive app IDs in those URIs failing).  Requiring callers to pass `appId` if case-sensitivity is an issue seems like a reasonable starting point; if users find they need the case-sensitivity with the more generic case, then we could look at supporting that.  (An alternative is, always, to *not* use case-sensitive app IDs in the first place.)

All that to say, I'd prefer to avoid all the URL parsing can of worms just to pick out the original host, if there's no significant demand for that use case.